### DISCUSSION TITLE
Does garbage collection affect the resources not installed by fluxcd?

### DISCUSSION THREAD

**Discussion by abisare** [fluxcd/flux2#2901] [Q&A] [answered]
Hi
I'm working on migration of helm releases installed by helm to be installed by flux v2. 
as per [garbage collection documentation](https://fluxcd.io/docs/components/kustomize/kustomization/#garbage-collection), previously added resources will be removed from the cluster if they're no longer present in the git repository. 
Will garbage collector remove the resources that do not belong to flux installation?
Does kustomize controller look at labels that were added during installation with flux to know what to delete? (i.e helm.toolkit.fluxcd.io/name: <name>, helm.toolkit.fluxcd.io/namespace: <ns>).
I'm pretty sure the resources will be kept, but I want to make sure that nothing undesired will be deleted from my cluster. 
Thanks in advance.

**Comment by kingdonb** [comment, marked as answer]
No, Flux will only remove resources that were in the Kustomization inventory. Flux used to use labels to facilitate garbage collection and it was possible for Flux to inadvertently remove resources that Flux did not create if they had labels that matched Flux's garbage collection. This is no longer possible since about Flux 0.18, when server-side apply was introduced we now use an Inventory system.

Flux will only garbage collect resources that it created.

**Reply by abisare** [reply]
Great to hear that! Thanks for reply, appreciate it!

**Reply by mathias-muench** [reply]
It may be worth commenting that resources not installed by fluxcd may be removed in an "indirect" way. For example: When garbage collection removes a namespace, it also removes serviceaccounts within that namespace that were originally created by `eksctl`.
